packages/lib/src/core/Analytics/utils.ts

```diff
@@ -139,19 +140,17 @@ export const getCardConfigData = (cardProps: CardConfiguration): CardConfigData
         showKCPType = countryCode?.toLowerCase() === 'kr' ? 'atStart' : 'auto';
     }
 
-    // Probably just for development - in real life we wouldn't expect the number of supported brands to push the endpoint limit on 128 chars
-    let brandsStr = brands.toString();
-    if (brandsStr.length > 128) {
-        brandsStr = brandsStr.substring(0, 124);
-    }
     // @ts-ignore commenting out props until endpoint is ready
     const configData: CardConfigData = {
         autoFocus,
-        ...(billingAddressAllowedCountries?.length > 0 && { billingAddressAllowedCountries: billingAddressAllowedCountries.toString() }),
+        ...(billingAddressAllowedCountries?.length > 0 && {
+            billingAddressAllowedCountries: billingAddressAllowedCountries.toString().substring(0, MAX_LENGTH)
```

Comment:
Does this also need the optional chaining operator (`?`) after `toString()`
(You used it for `billingAddressRequiredFields` and `brands`)

Reply:
It should be fine. The engine checks `billingAddressAllowedCountries?.length > 0 ` first, if it failed, it won't execute `billingAddressAllowedCountries: billingAddressAllowedCountries.toString().substring(0, MAX_LENGTH)`, so it shouldn't cause any exceptions.